persist-search-term-staged-rollout-phase-1.toml

```diff
@@ -0,0 +1,100 @@
+data_source = "main_search_counts"
+select_expression = "COALESCE(SUM(urlbar_navigation_search_formhistory),0)"
+
+[metrics.urlbar_persist_navigation_search_suggestion]
+data_source = "main_search_counts"
+select_expression = "COALESCE(SUM(urlbar_navigation_search_suggestion),0)"
+
+[metrics.search_content_urlbar_persisted_tagged_search.statistics.bootstrap_mean]
+[metrics.search_content_urlbar_persisted_tagged_follow_on_search.statistics.bootstrap_mean]
+[metrics.persisted_adclicks.statistics.bootstrap_mean]
+[metrics.persisted_search_w_ads.statistics.bootstrap_mean]
+[metrics.urlbar_persist_navigation_search_enter.statistics.bootstrap_mean]
+[metrics.urlbar_persist_navigation_search_formhistory.statistics.bootstrap_mean]
+[metrics.urlbar_persist_navigation_search_suggestion.statistics.bootstrap_mean]
+
+
+[data_sources]
+
+[data_sources.main_search_counts]
+from_expression = """(
+  SELECT client_id,
+    DATE(submission_timestamp) AS submission_date,
+    environment.experiments,
+    CASE WHEN e.key LIKE 'google:tagged:%' THEN e.value ELSE 0 END AS google_tagged_searches,
+    CASE WHEN e.key LIKE 'bing:tagged:%' THEN e.value ELSE 0 END AS bing_tagged_searches,
+    CASE WHEN e.key LIKE 'duckduckgo:tagged:%' THEN e.value ELSE 0 END AS ddg_tagged_searches, 
+    CASE WHEN e.key LIKE 'google:tagged-fo%' THEN e.value ELSE 0 END AS google_tagged_follow_on_searches,
+    CASE WHEN e.key LIKE 'bing:tagged-fo%' THEN e.value ELSE 0 END AS bing_tagged_follow_on_searches,
+    CASE WHEN e.key LIKE 'duckduckgo:tagged-fo%' THEN e.value ELSE 0 END AS ddg_tagged_follow_on_searches, 
+    CASE WHEN b.key  = 'search_enter' THEN b.value ELSE 0 END AS urlbar_navigation_search_enter,
+    CASE WHEN b.key  = 'search_formhistory' THEN b.value ELSE 0 END AS urlbar_navigation_search_formhistory,
+    CASE WHEN b.key  = 'search_suggestion' THEN b.value ELSE 0 END AS urlbar_navigation_search_suggestion,
+    CASE WHEN a.key LIKE 'google:%' THEN a.value ELSE 0 END AS google_ad_clicks,
+    CASE WHEN a.key LIKE 'bing:%' THEN a.value ELSE 0 END AS bing_ad_clicks,
+    CASE WHEN a.key LIKE 'duckduckgo:%' THEN a.value ELSE 0 END AS ddg_ad_clicks, 
+    CASE WHEN s.key LIKE 'google:%' THEN s.value ELSE 0 END AS google_search_w_ad,
+    CASE WHEN s.key LIKE 'bing:%' THEN s.value ELSE 0 END AS bing_search_w_ad,
+    CASE WHEN s.key LIKE 'duckduckgo:%' THEN s.value ELSE 0 END AS ddg_search_w_ad
+  FROM `moz-fx-data-shared-prod.telemetry_stable.main_v4` 
+    CROSS JOIN UNNEST(payload.processes.parent.keyed_scalars.browser_search_content_urlbar_persisted) e,
+    UNNEST(payload.processes.parent.keyed_scalars.browser_engagement_navigation_urlbar_persisted) b,
+    UNNEST(payload.processes.parent.keyed_scalars.browser_search_adclicks_urlbar_persisted) a,
+    UNNEST(payload.processes.parent.keyed_scalars.browser_search_withads_urlbar_persisted) s
+  WHERE DATE(submission_timestamp) >= '2023-01-01' AND e.key IS NOT NULL
+)"""
+experiments_column_type = "native"
```

Comment:
This is correct.

There are a couple of types:



* ’simple’: There is an experiments column, which is an (experiment_slug:str -> branch_name:str) map.
* ’native’: There is an experiments column, which is an (experiment_slug:str -> struct) map, where the struct contains a branch field, which is the branch as a string.
*  None: There is no experiments column, so skip the sanity checks that rely on it. We’ll also be unable to filter out pre-enrollment data from day 0 in the experiment.